pkgs/servers/x11/xorg/overrides.nix

```diff
@@ -760,6 +760,9 @@ self: super:
     prePatch = ''
       sed -i 's|^defaultserverargs="|&-logfile \"$HOME/.xorg.log\"|p' startx.cpp
     '';
+    postFixup = ''
+      substituteInPlace $out/bin/startx --replace $out/etc/X11/xinit/xserverrc /etc/X11/xinit/xserverrc
+    '';
```

Comment:
So there is a default for startx, which is not effected by the default set in build? - In the commits coming after this?

Reply:
The default xserverrc is in the Nix store, this changes it to /etc/.

The following patches handle the logfile, not xserverrc. Technically I could've put this in a separate PR but since I was touching startx anyways, I thought I'd bundle them together.